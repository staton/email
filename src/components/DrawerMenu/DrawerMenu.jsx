import React, { Component } from 'react';
import ComposeEmailButton from '../ComposeEmailButton/ComposeEmailButton';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

export class DrawerMenu extends Component {

	render() {
		return (
            <div className="DrawerMenu">
                {
                    // For small screens, do not show the New Email button here.
                    // (It will be a FAB for small screens. (see: App.jsx))
                    (this.props.isSmallScreen)
                    ?   null
                    :   <ComposeEmailButton content="New Email" />
                }
                <ul>
                    <li>Menu Item 1</li>
                    <li>Menu Item 2</li>
                    <li>Menu Item 3</li>
                </ul>
            </div>
		);
    }

}

function mapStateToProps(store, ownProps) {
    return {
        isSmallScreen: store.app.isSmallScreen
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
    },
    dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(DrawerMenu);