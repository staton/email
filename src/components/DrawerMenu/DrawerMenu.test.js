import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { DrawerMenu } from './DrawerMenu';

describe('<DrawerMenu />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<DrawerMenu />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});