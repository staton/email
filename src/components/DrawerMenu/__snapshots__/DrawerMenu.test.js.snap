// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerMenu /> renders correctly 1`] = `
<div
  className="DrawerMenu"
>
  <ComposeEmailButton
    content="New Email"
    onClick={[Function]}
    title=""
  />
  <ul>
    <li>
      Menu Item 1
    </li>
    <li>
      Menu Item 2
    </li>
    <li>
      Menu Item 3
    </li>
  </ul>
</div>
`;
